### PR TITLE
DEVOPS-000: Missing image on install

### PR DESCRIPTION
**Pull request by sherakama**
# READY FOR REVIEW

# Summary
- Fixes the missing source image on install
- Improves the SQL Constraint error handling (by avoiding it) as PHP has a hard time catching it.

# Review By (Date)
- Today!

# Criticality
- 10 / 10
- Hotfix for release.

# Review Tasks

## Setup tasks and/or behavior to test

1. Review code
2. Check out this branch
3. Do a full site install `blt di` on the cardinalsites stack.

## Front End Validation
- [ ] Is the markup using the appropriate semantic tags and passes HTML validation?
- [ ] Cross-browser testing has been performed?
- [ ] Automated accessibility scans performed?
- [ ] Manual accessibility tests performed?
- [ ] Design is approved by @ user?

### Code
- [ ] Are the naming conventions following our standards?
- [ ] Does the code have sufficient inline comments?
- [ ] Is there anything in this code that would be hidden or hard to discover through the UI?
- [ ] Are there any [code smells](https://blog.codinghorror.com/code-smells/)?
- [ ] Are tests provided? eg (unit, behat, or codeception)

### Code security
- [ ] Are all [forms properly sanitized](https://www.drupal.org/docs/8/security/drupal-8-sanitizing-output)?
- [ ] Any obvious [security flaws or new areas for attack](https://www.drupal.org/docs/8/security)?

## General
- [ ] Is there anything included in this PR that is not related to the problem it is trying to solve?
- [ ] Is the approach to the problem appropriate?

# Affected Projects or Products
- Cardinalsites / LelandD8 / Cardinal Service

# Associated Issues and/or People
- D8CORE-2099
- @pookmish 

# Resources
- [AMP Tool](https://stanford.levelaccess.net/index.php)
- [Accessibility Manual Test Script](https://docs.google.com/document/d/1ZXJ9RIUNXsS674ow9j3qJ2g1OAkCjmqMXl0Gs8XHEPQ/edit?usp=sharing)
- [HTML Validator](https://validator.w3.org/)
- [Browserstack](https://live.browserstack.com/dashboard) and link to [Browserstack Credentials](https://asconfluence.stanford.edu/confluence/display/SWS/External+Account+Credentials)

### CODE REVIEW

File: stanford_person.install [hunk 1]
```diff
@@ -25,4 +25,9 @@ function stanford_person_install() {
   if (!\Drupal::service('module_handler')->moduleExists('default_content')) {
     _stanford_person_install_default_photo();
   }
+  else {
+    // Just put the file in place.
+    _stanford_person_install_default_photo_file();
```
Comment (pookmish) [inline stanford_person.install:30]: oh so this is for when the default content json is consumed, the image isn't created so we have to put it there ourself.

**Comment by pookmish** [comment]
should we change the merge branch to go into `master` so it'll cut a new tag. like a hotfix?

**Comment by imonroe** [comment]
That's the easy way; I don't have to run the script to cut the release, but if we do that, could we also update the changelog in this PR?

**Comment by pookmish** [comment]
and version number :point_up:

**Comment by imonroe** [comment]
Ok, I set it up with a changelog entry and I bumped the version in the info.yml.  It's also set to merge to master.  Once this tests green, I'll let @pookmish push the big green button.

File: composer.json [hunk 2]
```diff
@@ -95,9 +95,9 @@
         "drupal/views_infinite_scroll": "^1.6",
         "drupal/views_taxonomy_term_name_depth": "^6.0",
         "sainsburys/guzzle-oauth2-plugin": "^3.0",
-        "su-sws/jumpstart_ui": "^8.1.6",
-        "su-sws/stanford_migrate": "^8.1.0",
-        "su-sws/stanford_text_editor": "^8.1.6"
+        "su-sws/jumpstart_ui": "dev-8.x-1.x",
+        "su-sws/stanford_migrate": "dev-8.x-1.x",
+        "su-sws/stanford_text_editor": "dev-8.x-1.x"
```
Comment (pookmish) [inline composer.json:100]: gotta change these

File: templates/sections/section-person-quote.html.twig [hunk 3]
```diff
@@ -5,10 +5,10 @@
 #}
 
 {%- set attributes = attributes.addClass(['centered-container', 'layout--section-person-header']) -%}
-{%- set figure_attributes = create_attribute() -%}
+{%- set quote_attributes = create_attribute() -%}
```
Comment (pookmish) [inline templates/sections/section-person-quote.html.twig:8]: where this come from?

File: stanford_person.install (hunk 1 above)
Comment (sherakama) [reply stanford_person.install:30]: Yeah, that was my learning moment as well. Got tricked by the fact my development environment had the image already there.

File: templates/sections/section-person-quote.html.twig (hunk 3 above)
Comment (sherakama) [reply templates/sections/section-person-quote.html.twig:8]: Post-release merge. Came in after last tag.

Comment (sherakama) [reply templates/sections/section-person-quote.html.twig:8]: Probably should have branched off of master if we wanted to do a right hotfix. I could cherry pick it.

File: composer.json (hunk 2 above)
Comment (imonroe) [reply composer.json]: Fixed it.

File: templates/sections/section-person-quote.html.twig (hunk 3 above)
Comment (pookmish) [reply templates/sections/section-person-quote.html.twig:8]: meh

**Pull request merged**